rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json

```
@@ -188,26 +252,94 @@
         "source": {
           "title": "Source",
           "type": "string",
-          "minLength": 3
+          "minLength": 3,
+          "form": {
+            "expressionProperties": {
+              "templateOptions.required": "field.parent.model.url !== undefined && field.parent.model.url !== ''"
+            }
+          }
         },
         "url": {
           "title": "URL",
           "type": "string",
           "placeholder": "Example: https://www.rero.ch/",
           "format": "uri",
-          "minLength": 7
+          "minLength": 7,
+          "form": {
+            "expressionProperties": {
+              "templateOptions.required": "field.parent.model.source !== undefined && field.parent.model.source !== ''"
+            }
+          }
         }
       }
     },
-    "note": {
-      "title": "Note",
-      "description": "Add further information to the librarian.",
-      "type": "string",
-      "maxLength": 2000,
+    "notes": {
+      "title": "Notes",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "additionalProperties": false,
+        "title": "Note",
+        "propertiesOrder": [
+          "type",
+          "content"
+        ],
+        "required": [
+          "type",
+          "content"
+        ],
+        "properties": {
+          "type": {
+            "type": "string",
+            "title": "Type",
```

Properties order (in the schema, not for the editor):
```suggestion
            "title": "Type",
            "type": "string",
```

---

done